### PR TITLE
Document desktop role options for Teleport 9

### LINKED ISSUE #10160: Document new role options for desktop access
## Details

For Teleport 9, we've added:

```
options:
  record_sessions:
    desktop: true
  desktop_clipboard: true
```

Document these new options.

Both of them default to true if not specified, and attempt to lean towards the _most secure_ option.

 - For session recording, the most secure thing is to record, so in order to disable recording **all** of a user's roles must explicitly disable it.

 - For clipboard sharing, the most secure thing is to not share clipboards, so in order to disable clipboard sharing, **one or more** or a user's roles must explicitly disable it.

More context in [this PR comment](https://github.com/gravitational/teleport/pull/9868#discussion_r791978457), when we implement clipboard support for Desktop Access, we should document the behavior of the [`desktop_clipboard` role option](https://github.com/gravitational/teleport/blame/080ba8fc326fd500a29023aa87c51ff1f38f7ff3/rfd/0049-desktop-clipboard.md#L134-L139) per RFD 49:

> For these reasons, clipboard access will be enabled via the `desktop_clipboard`
> role option as specified in [RFD 33](./0033-desktop-access.md). This option will
> default to `true`, but the presence of a single role that disables the clipboard
> will turn it off. When clipboard access is not enabled, Teleport will not
> respond to any clipboard messages received by either the browser or the RDP
> server.

### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by zmb3**
Fixes #10160

No backport required, these are specific to Teleport 9+

### CODE REVIEW

File: docs/pages/access-controls/reference.mdx [hunk 1]
```diff
@@ -72,6 +72,17 @@ spec:
     max_sessions: 10
     # permit_x11_forwarding allows users to use X11 forwarding with openssh clients and servers through the proxy
     permit_x11_forwarding: true
+    # Specify whether or not to record the user's desktop sessions.
+    # The cluster's session recording mode must not be set to "off".
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:76]: Should we mention `auth_service.session_recording` specifically here?

File: docs/pages/access-controls/reference.mdx [hunk 2]
```diff
@@ -81,7 +92,7 @@ spec:
     logins: [root, '{{internal.logins}}']
     # Windows logins a user is allowed to use for desktop sessions.
     windows_desktop_logins: [Administrator, '{{internal.logins}}']
-    # If kubernetes integration is enabled, this setting configures which 
+    # If kubernetes integration is enabled, this setting configures which
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:95]:
```suggestion
    # If the Kubernetes integration is enabled, this setting configures which
```

File: docs/pages/desktop-access/reference.mdx [hunk 3]
```diff
@@ -64,6 +64,19 @@ version: v4
 metadata:
   name: developer
 spec:
+  options:
+    # Specify whether or not to record the user's desktop sessions.
+    # The cluster's session recording mode must not be set to "off".
```
Comment (ptgott) [inline docs/pages/desktop-access/reference.mdx:69]: Should we mention `auth_service.session_recording` specifically here?

File: docs/pages/setup/reference/resources.mdx [hunk 4]
```diff
@@ -211,6 +211,17 @@ spec:
     # cluster. This setting slows down Teleport performance because it has to track
     # connections cluster-wide.
     max_connections: 2
+    # Specify whether or not to record the user's desktop sessions.
+    # The cluster's session recording mode must not be set to "off".
```
Comment (ptgott) [inline docs/pages/setup/reference/resources.mdx:215]: Should we mention `auth_service.session_recording` specifically here?

File: docs/pages/access-controls/reference.mdx (hunk 1 above)
Comment (zmb3) [reply docs/pages/access-controls/reference.mdx:76]: We could, it's either there for static config, or in the `SessionRecordingConfig` for dynamic config.

Comment (ptgott) [reply docs/pages/access-controls/reference.mdx:76]: Oh right. Maybe we could add a two-item unordered list after this sentence with the specific config fields?

Comment (zmb3) [reply docs/pages/access-controls/reference.mdx]: Take a look at the latest commit and LMK what you think.

**Review by ptgott — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**